src/ape/contracts/base.py

```diff
@@ -677,6 +677,36 @@ def _mutable_methods_(self) -> Dict[str, ContractTransactionHandler]:
             # NOTE: Must raise AttributeError for __attr__ method or will seg fault
             raise AttributeError(str(err)) from err
 
+    def method_call(self, method_name: str, *args, **kwargs) -> Any:
+        """
+        Call a contract function directly using the method_name.
+        This is helpful in the scenario where the contract has a
+        method name matching an attribute of the BaseAddress class,
+        such as 'nonce' or 'balance'
+
+        Args:
+            method_name (str): The contract method name to be called
+
+        Returns:
+            Output of smart contract interaction
+
+        """
+        if method_name not in {*self._view_methods_, *self._mutable_methods_}:
```

Comment:
This section could be an `else:` after the next set if `if / elif` to save a comparison